flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManagerDriver.java

```diff
@@ -228,21 +221,27 @@ public void deregisterApplication(ApplicationStatus finalStatus, @Nullable Strin
 
 	@Override
 	public CompletableFuture<YarnWorkerNode> requestResource(TaskExecutorProcessSpec taskExecutorProcessSpec) {
-		final Optional<Resource> containerResourceOptional = getContainerResource(taskExecutorProcessSpec);
+		checkInitialized();
+
 		final CompletableFuture<YarnWorkerNode> requestResourceFuture = new CompletableFuture<>();
 
-		if (containerResourceOptional.isPresent()) {
-			resourceManagerClient.addContainerRequest(getContainerRequest(containerResourceOptional.get()));
+		final Optional<TaskExecutorProcessSpecContainerResourcePriorityAdapter.PriorityAndResource> priorityAndResourceOpt =
+			taskExecutorProcessSpecContainerResourcePriorityAdapter.getPriorityAndResource(taskExecutorProcessSpec);
+
+		if (!priorityAndResourceOpt.isPresent()) {
+			requestResourceFuture.completeExceptionally(
+				new ResourceManagerException(String.format("Could not compute the container Resource from the given TaskExecutorProcessSpec %s.", taskExecutorProcessSpec)));
```

Comment:
There should be other logs from `taskExecutorProcessSpecContainerResourcePriorityAdapter` that gives the reasons why resource cannot be computed. For the moment, the only possible reason is that the requested resource is larger than the max container resource Yarn allows. See `TaskExecutorProcessSpecContainerResourcePriorityAdapter#tryAdaptResource` for the logs.

Reply:
True the logs exist. However, it might be not that easy to correlate these logging statements with the logging of the `ResourceManagerException` at a different place. If there is currently only one reason why the operation fails, maybe we can state it in the exception message as well?